### PR TITLE
Feature/autosavings

### PR DESCRIPTION
**Pull request by zeroknots**

### CODE REVIEW

File: src/executors/AutoSavings.sol [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        bytes32 id,
+        uint256 amountTransfered,
+        ConditionConfig[] calldata conditions
+    )
+        external
+        onlyIfConditionsMet(msg.sender, conditions)
+    {
+        SavingsConfig storage config = savingsConfig[msg.sender][id];
+
+        uint256 amountIn = config.feePercentage * amountTransfered / 10_000;
```
Comment (kopy-kat) [inline src/executors/AutoSavings.sol:75]: why is this divided by 10,000?

File: src/validators/SessionKey/ERC20SessionKeyValidator.sol [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            }
+            if (callValue > 0) {
+                revert("ERC20SV Non Zero Value");
+            }
+        }
+        // working with userOp.callData
+        // check if the call is to the allowed recepient and amount is not more than allowed
+        bytes calldata data;
+        {
+            uint256 offset = uint256(bytes32(_op.callData[4 + 64:4 + 96]));
```
Comment (kopy-kat) [inline src/validators/SessionKey/ERC20SessionKeyValidator.sol]: do you think it'd be more readable to use 68:100 and explain the calculation in comments?

File: src/validators/SessionKey/ERC20SessionKeyValidator.sol [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            uint256 length = uint256(bytes32(_op.callData[4 + offset:4 + offset + 32]));
+            //we expect data to be the `IERC20.transfer` calldata
+            data = _op.callData[4 + offset + 32:4 + offset + 32 + length];
```
Comment (kopy-kat) [inline src/validators/SessionKey/ERC20SessionKeyValidator.sol]: same as above

File: src/validators/SessionKey/SessionKeyManager.sol [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    function isValidSignature(
+        bytes32 _dataHash,
+        bytes memory _signature
+    )
+        public
+        view
+        override
+        returns (bytes4)
+    {
+        return 0xffffffff; // do not support it here
```
Comment (kopy-kat) [inline src/validators/SessionKey/SessionKeyManager.sol:124]: Should we route this to the session key module?

File: src/executors/AutoSavings.sol (hunk 1 above)
Comment (zeroknots) [reply src/executors/AutoSavings.sol:75]: so you can have percentages sub 1%

File: src/validators/SessionKey/ERC20SessionKeyValidator.sol (hunk 2 above)
Comment (zeroknots) [reply src/validators/SessionKey/ERC20SessionKeyValidator.sol]: yea. think were just copying biconomies here

File: src/validators/SessionKey/SessionKeyManager.sol (hunk 4 above)
Comment (zeroknots) [reply src/validators/SessionKey/SessionKeyManager.sol:124]: i think the the forwarding/routing should take place in the signature validation manager that handles ERC1271

File: src/validators/SessionKey/ERC20SessionKeyValidator.sol (hunk 2 above)
Comment (zeroknots) [reply src/validators/SessionKey/ERC20SessionKeyValidator.sol]: https://github.com/rhinestonewtf/modulekit-examples/pull/5/commits/9e677bd6a3f981c6eff17414394f46372d935fb3

File: src/validators/SessionKey/ERC20SessionKeyValidator.sol (hunk 3 above)
Comment (zeroknots) [reply src/validators/SessionKey/ERC20SessionKeyValidator.sol]: https://github.com/rhinestonewtf/modulekit-examples/pull/5/commits/9e677bd6a3f981c6eff17414394f46372d935fb3

**Review by zeroknots — COMMENTED** [review]
Sweet looks good

**Pull request merged**